### ISSUE TITLE
Add race sounds

### ISSUE THREAD

**Issue by oy** [teeworlds/teeworlds#2388]
Add sounds for events (start, finish, record, checkpoint?) in race.

**Comment by heinrich5991** [comment]
Please design non-intrusive sounds for start and checkpoint at least, speed runners and other people playing races are going to hear them a lot.

I think TrackMania has a simple "click" sound on checkpoints, and a starting gun sound for the start. Not sure what they do for the finish.

**Comment by Dune-jr** [comment]
I agree that a simple sound like a single note would be for frequently heard sounds

**Comment by Fisico** [comment]
@oy 
I already did some a few weeks ago, still in testing but I would like to have following sound triggers: Some are not possible due to needed protocol changes according to @axblk.
Checkpoint server best (needs protocol change) and racemod/client implementation (purple color)
Checkpoint personal best
Checkpoint slower 
Race timer start
Finish – server best 
Finish - personal best
Finish - slower

What I think is important to add is a possibility in code and/or config to adjust the volume of each sample individually, it is very tedious to export and convert them over and over again, just to make them 1-2lufs quieter or louder. 

What I have come up with already still in testing.... Feedback please!


[Fisico_TW_Sounds.zip](https://github.com/teeworlds/teeworlds/files/4046884/Fisico_TW_Sounds.zip)

**Comment by oy** [comment]
Yeah, that list seems good.
That current start sound is kinda loud, could be smoother.

**Comment by Fisico** [comment]
> Yeah, that list seems good.
> That current start sound is kinda loud, could be smoother.

Have you tried it ingame?
Do you want a quieter sound or a software sound with less transient? Tbh most TW sounds are very harsh. For example the hook and all gun sounds.
It is way quieter than the hook for example.

I designed it in a way that you can hear it over and over again and it is not tiring for your brain because it has not very harsh frequencies. Sure if you stand on a checkpoint for example it will loop the whole time and after some time it will get annoying. But this can be solved in the code, only loop checkpoint sounds every 5 seconds for example. 

But as I said, a way to change the volume of the sample would be very handy.

**Comment by Fisico** [comment]
@Dune-jr updated his Gamer Client with the sounds to try them out.
https://www.teeworlds.com/forum/viewtopic.php?pid=123916#p123916

**Comment by Dune-jr** [comment]
I cherry picked from https://github.com/LordSk/teeworlds/commits/fisico_sound (and updated the sounds).

**Comment by Dune-jr** [comment]
@Fisico The "bad" and "good" checkpoints sound mixed up. The bad ones are those you hear most often, so surely they should be the least noisy/most discreet?

**Comment by Fisico** [comment]
> @Fisico The "bad" and "good" checkpoints sound mixed up. The bad ones are those you hear most often, so surely they should be the least noisy/most discreet?

Did you use @LordSk's code for the checkpoints? He made an error and used the checkpoint_faster sound for slower times and vice versa.
The files i sent you are correct.

**Comment by Dune-jr** [comment]
> Did you use @LordSk's code for the checkpoints? He made an error and used the checkpoint_faster sound for slower times and vice versa.

Yes. Makes sense then
